plugins/statistics.koplugin/calendarview.lua

```diff
@@ -364,6 +365,609 @@ function CalendarWeek:update()
+        self.footer_first_up:enableDisable(self.show_page > 1)
+        self.footer_last_down:enableDisable(self.show_page < self.pages)
+
+        self.timeline_height = self.timeline_height - self.footer_height
+    else
+        self.timeline_height = self.timeline_height - Size.padding.default
+    end
+    self.hour_width = math.floor(self.timeline_height / 24)
+    self.timeline_width = self.dimen.w - self.outer_padding - self.time_text_width
+
+    if #self.kv_pairs == 0 then
+        -- Needed when the first opened day has no data, then move to another day with data
+        table.insert(self.book_items, CenterContainer:new{
+            dimen = Geom:new { w = self.dimen.w - 2 * self.outer_padding, h = 0},
+            VerticalSpan:new{ w = 0 }
+        })
+    end
+    self:refreshTimeline()
+
+end
+
+function CalendarDayView:refreshTimeline()
+    self.timeline:clear()
+    for _, v in ipairs(self.kv_pairs) do
+        if v.checked and v.periods then
+            local fgcolor, bgcolor = unpack(SPAN_COLORS[(v.book_id % #SPAN_COLORS)+1])
+            for _, period in ipairs(v.periods) do
+                local start_hour = math.floor(period.start / 3600)
+                local finish_hour = math.floor(period.finish / 3600)
+                for i=0, finish_hour-start_hour do
+                    local start = i==0 and period.start or (start_hour+i) * 3600
+                    if start >= 24 * 3600 then
+                        break
+                    end
+                    local finish = i==finish_hour-start_hour and period.finish or (start_hour+i+1) * 3600 - 1
+                    local span = self:generateSpan(start, finish, bgcolor, fgcolor, v[1])
+                    if span then table.insert(self.timeline, span) end
+                end
+            end
+        end
+    end
+
+    for i=0, 23 do
+        local offset_y = self.timeline_offset + self.hour_width * i
+        table.insert(self.timeline, FrameContainer:new{
+            width = self.time_text_width,
+            height = self.hour_width,
+            margin = 0,
+            padding = 0,
+            background = Blitbuffer.COLOR_WHITE,
+            bordersize = 0,
+            overlap_offset = {0, offset_y},
+            CenterContainer:new{
+                dimen = Geom:new{ w = self.time_text_width, h = self.hour_width},
+                TextWidget:new{
+                    text = string.format("%02d:00", i),
+                    face = self.time_text_face,
+                    padding = Size.padding.small
+                }
+            }
+        })
+
+        if i ~= 0 then
+            table.insert(self.timeline, FrameContainer:new{
+                width = self.timeline_width,
+                height = Size.border.thin,
+                background = Blitbuffer.COLOR_LIGHT_GRAY,
+                bordersize = 0,
+                padding = 0,
+                overlap_offset = { self.time_text_width, offset_y - Size.border.thin/2 },
```

Comment:
MIght want to add it to Size as `Size.border.thinner` or `Size.border.extra_thin`.
Also, math.floor() if doing that here - we absolutely never want non-integer x/y/w/h, or you're in for a hell of debugging positionning issues :)

Was that your "Done as good as I can, without using thicker lines." ? May be these non-integer values were the cause?
Also, shouldn't you better fix the height of the spans to be `height of the line - separator - 2*some_padding`, to solve thie issue - or then the text is not vertically aligned in the span (althought this might be less noticable?)